homeassistant/components/xiaomi_miio/select.py

```diff
@@ -79,45 +150,30 @@ async def async_setup_entry(
     if not config_entry.data[CONF_FLOW_TYPE] == CONF_DEVICE:
         return
 
-    entities = []
     model = config_entry.data[CONF_MODEL]
-    device = hass.data[DOMAIN][config_entry.entry_id][KEY_DEVICE]
-
-    if model == MODEL_AIRPURIFIER_3C:
-        return
-    if model in MODELS_HUMIDIFIER_MIIO:
-        entity_class = XiaomiAirHumidifierSelector
-    elif model in MODELS_HUMIDIFIER_MIOT:
-        entity_class = XiaomiAirHumidifierMiotSelector
-    elif model in [MODEL_AIRPURIFIER_M1, MODEL_AIRPURIFIER_M2]:
-        entity_class = XiaomiAirPurifierSelector
-    elif model in MODELS_PURIFIER_MIOT:
-        entity_class = XiaomiAirPurifierMiotSelector
-    elif model == MODEL_AIRFRESH_VA2:
-        entity_class = XiaomiAirFreshSelector
-    elif model in (
-        MODEL_FAN_ZA1,
-        MODEL_FAN_ZA3,
-        MODEL_FAN_ZA4,
-        MODEL_FAN_SA1,
-        MODEL_FAN_V2,
-        MODEL_FAN_V3,
-    ):
-        entity_class = XiaomiFanSelector
-    else:
+    if model not in MODEL_TO_ATTR_MAP:
         return
 
-    description = SELECTOR_TYPES[FEATURE_SET_LED_BRIGHTNESS]
-    entities.append(
-        entity_class(
-            f"{config_entry.title} {description.name}",
-            device,
-            config_entry,
-            f"{description.key}_{config_entry.unique_id}",
-            hass.data[DOMAIN][config_entry.entry_id][KEY_COORDINATOR],
-            description,
-        )
-    )
+    entities = []
+    unique_id = config_entry.unique_id
+    device = hass.data[DOMAIN][config_entry.entry_id][KEY_DEVICE]
+    coordinator = hass.data[DOMAIN][config_entry.entry_id][KEY_COORDINATOR]
+    attributes = MODEL_TO_ATTR_MAP[model]
+
+    for description in SELECTOR_TYPES:
+        for attribute in attributes:
+            if description.key == attribute[0]:
```

Comment:
Use named attribute access for the value here. See my comment above for namedtuples.